docs/ocis/adr/0020-file-search-query-language.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "20. File Search Query Language"
+date: 2022-03-18T09:00:00+01:00
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0018-file-search-query-language.md
+---
+
+* Status: proposed
+* Deciders: @butonic, @micbar, @dragotin, @C0rby
+* Date: 2022-03-18
+
+## Context and Problem Statement
+
+From the users perspective, the interface to search is just a single form field where the user enters one or more search terms. The minimum expectation is that the search returns file names and links to files that
+
+* have a file name that contains at least one of the search terms
+* contain at least one of the search terms in the file contents
+* have meta data that is equal or contains one of the search terms
+
+## Decision Drivers
+
+* The standard user should not be bothered by a query syntax
+* The power user should also be able to narrow his search with an efficient and flexible syntax
+* We need to consider different backend technologies which we need to access through an abstraction layer
+* Using different indexing systems should lead to a slightly different feature set whitout changing the syntax completely
+
+## Considered Options
+
+* [Keyword Query Language](#keyword-query-language-kql)
+* [Simple Query](#simplified-query)
+* [Lucene Query Language](#lucene-query-language)
+* [Solr Query Language](#solr-query-language)
+* [Elasticsearch Query Language](#elasticsearch-query-language)
+
+## Decision Outcome
+
+Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+## Pros and Cons of the Options
+
+### Keyword Query Language (KQL)
+
+Implement a search based on the Keyword Query Language (KQL), adopted from Sharepoint.
+
+* Good, because microsoft already uses it togethe with the GraphAPI
```

Comment:
I am not sure if this is true. The _Keyword Query Language_ comes with Sharepoint. For Graph, I could only find references to _Kusto Query Language_ which is more for data. Do I overlook something? Confusing...

Reply:
ok, why did you pick the KQL? what is on the "good" side?

Reply:
removed it, could not find any good reasons.